src/components/Metatags/Metatags.tsx

```diff
@@ -0,0 +1,187 @@
+import { IDocsEntity } from '@api';
+import React from 'react';
+import Head from 'next/head';
+import { Esources } from '@api/lib/search/types';
+
+const ADS_BASE_URL = 'https://ui.adsabs.harvard.edu';
+
+const ABS_BASE_URL = `${ADS_BASE_URL}/abs`;
+
+const LINKGWAY_BASE_URL = `${ADS_BASE_URL}/link_gateway`;
+interface IMetatagsProps {
+  doc: IDocsEntity;
+}
+
+export const metatagsQueryFields: Partial<keyof IDocsEntity>[] = [
+  'abstract',
+  'aff',
+  'author',
+  'bibcode',
+  'bibstem',
+  'doctype',
+  'doi',
+  'esources',
+  'identifier',
+  'isbn',
+  'issn',
+  'issue',
+  'keyword',
+  'page',
+  'page_range',
+  'pub',
+  'pubdate',
+  'title',
+  'volume',
+];
+
+export const Metatags = (props: IMetatagsProps): React.ReactElement => {
+  const { doc } = props;
+
+  if (!doc) {
+    return null;
+  }
+
+  const getFomattedNumericPubdate = (pubdate: string) => {
+    const regex = /(....)-(..)-(..)/;
+    const res = regex.exec(pubdate);
+    return `${res[1]}/${res[2]}`;
+  };
+
+  const getLastPage = (page_range: string) => {
+    const pages = page_range.split('-');
+    return pages[1];
+  };
+
+  const getArXiv = (identifier: string[]) => {
+    identifier.forEach((id) => {
+      if (id.startsWith('arXiv')) {
+        return id;
+      }
+    });
+    return '';
+  };
```

Comment:
Not sure about this function, `forEach` won't return anything, assuming you're looking for `filter`?